### ISSUE TITLE
1.7: duplicate OUTPUT rules on restart

### ISSUE THREAD

**Issue by ibuildthecloud** [docker/docker#13956]
Each time you restart the Docker daemon another OUTPUT rule gets added in iptables.  After a bunch of restarts I have 

```
Chain OUTPUT (policy ACCEPT 10 packets, 600 bytes)
 pkts bytes target     prot opt in     out     source               destination         
25932 1603K DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL
20553 1280K DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL
20482 1276K DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL
20191 1258K DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL
20144 1255K DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL
20030 1248K DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL
19897 1240K DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL
19659 1225K DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL
19632 1224K DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL
19603 1222K DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL
19574 1220K DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL
19574 1220K DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL
19048 1185K DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL
18945 1179K DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL
```

**Comment by ibuildthecloud** [comment]
Tested against 1.7.0-rc4

**Comment by ibuildthecloud** [comment]
@mavenugo

**Comment by ibuildthecloud** [comment]
@icecrime @jfrazelle should this be fixed??

**Comment by LK4D4** [comment]
@ibuildthecloud I think should be.

**Comment by jfrazelle** [comment]
Yes it should be and this is definitely a regression

**Comment by ibuildthecloud** [comment]
Somewhat related, maybe less impactful but caused issues for me => https://github.com/docker/docker/issues/13957

**Comment by unclejack** [comment]
Can we also get some tests along with the fix for this issue, please? It has been a problem in the past so we should really test for this.

**Comment by icecrime** [comment]
Looking into this: test here is always false and the rule gets recreated https://github.com/docker/libnetwork/blob/master/iptables/iptables.go#L114

**Comment by icecrime** [comment]
The `Exists` test using `-C` fails with:

    iptables: Bad rule (does a matching rule exist in that chain?).

We end up in the "long path" where we test if the output of `/sbin/iptables -t nat -S OUTPUT` contains `""-m addrtype --dst-type LOCAL ! --dst 127.0.0.0/8"`. In my case, the position of the `!` results in `false`:

```
/sbin/iptables -t nat -S OUTPUT
-P OUTPUT ACCEPT
-A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER
```

**Comment by icecrime** [comment]
Will close with vendoring bump.

**Comment by icecrime** [comment]
Nevermind: closed by docker/libnetwork#299.
